app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt

```diff
@@ -103,59 +117,65 @@ class RecFragment(contentLayoutId: Int) : Fragment(contentLayoutId) {
         return requireContext().externalCacheDir?.absolutePath + "/recording.3gp"
     }
 
+    @SuppressLint("Permissions are checked before calling this method", "MissingPermission")
     private fun startRecording() {
-        //initialize the media recorder
-        mediaRecorder = MediaRecorder()
-        //set the audio source
-        mediaRecorder.setAudioSource(MediaRecorder.AudioSource.MIC)
-        //set the output format
-        mediaRecorder.setOutputFormat(MediaRecorder.OutputFormat.THREE_GPP)
-        //set the output file path
-        mediaRecorder.setOutputFile(getOutputFilePath())
-        //set the encoder
-        mediaRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AAC)
-
-        try {
-            //prepare the media recorder
-            mediaRecorder.prepare()
-            //start recording
-            mediaRecorder.start()
-            //set recording in progress
-            isRecording = true
-            //change the button text
-            recordButton.text = "Stop Recording"
-
-            //set the timer
-            timer = object : CountDownTimer(MILLISINFUTURE, COUNT_DOWN_INTERVAL) {
-                override fun onTick(millisUntilFinished: Long) {
-                    recordingTimeText.text = ((MILLISINFUTURE - millisUntilFinished) / COUNT_DOWN_INTERVAL).toString()
-                }
-                override fun onFinish() {
-                //do nothing
+        // Initialize the AudioRecord
+        audioRecorder = AudioRecord(audioSource, sampleRateInHz, channelConfig, audioFormat, bufferSize)
+
+        // Set the output file path
+        outputFilePath = getOutputFilePath()
+        outputFile = File(outputFilePath)
+
+        // Start recording
+        audioRecorder.startRecording()
+        isRecording = true
+        recordButton.text = "Stop Recording"
+
+        // Write the audio data to a file
+        val buffer = ByteArray(bufferSize)
+        val outputStream = FileOutputStream(outputFile)
+
+        Thread(Runnable {
+            while (isRecording) {
+                val bytesRead = audioRecorder.read(buffer, 0, bufferSize)
+                if (bytesRead > 0) {
+                    try {
+                        outputStream.write(buffer, 0, bytesRead)
+                    } catch (e: IOException) {
+                        e.printStackTrace()
+                    }
                 }
```

Comment:
if(bytesRead <= 0), should that be an error? Could add a toast as a small indication?